cmd/expiration-mailer/main.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2015 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/mail"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type emailContent struct {
+	ExpirationDate   time.Time
+	DaysToExpiration int
+	CommonName       string
+	DNSNames         string
+}
+
+type mailer struct {
+	stats         statsd.Statter
+	log           *blog.AuditLogger
+	dbMap         *gorp.DbMap
+	Mailer        mail.Mailer
+	EmailTemplate *template.Template
+	WarningDays   []int
```

Comment:
I'd call this WarningTimes or WarningDurations or similar, and make it a slice of Duration.